modules/solid_mechanics/src/materials/NewtonIterationSingleVariable.C

```diff
@@ -0,0 +1,443 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "NewtonIterationSingleVariable.h"
+
+#include "InputParameters.h"
+#include <cmath>
+#include "Conversion.h"
+
+template <>
+InputParameters
+validParams<NewtonIterationSingleVariable>()
+{
+  InputParameters params = emptyInputParameters();
+
+  // Newton iteration control parameters
+  params.addParam<unsigned int>("max_its", 200, "Maximum number of Newton iterations");
+  params.addParam<bool>(
+      "output_iteration_info", false, "Set true to output Newton iteration information");
+  params.addDeprecatedParam<bool>(
+      "output_iteration_info_on_error",
+      false,
+      "Set true to output Newton iteration information when those iterations fail",
+      "This information is always output when iterations fail");
+  params.addParam<Real>(
+      "relative_tolerance", 1e-8, "Relative convergence tolerance for Newton iteration");
+  params.addParam<Real>(
+      "absolute_tolerance", 1e-11, "Absolute convergence tolerance for Newton iteration");
+  params.addParam<bool>("legacy_return_mapping",
+                        false,
+                        "Perform iterations and check for residual "
+                        "the same way the same way as the previous "
+                        "algorithm");
+
+  return params;
+}
+
+NewtonIterationSingleVariable::NewtonIterationSingleVariable(const InputParameters & parameters)
+  : _legacy_residual(false),
+    _max_its(parameters.get<unsigned int>("max_its")),
+    _output_iteration_info(parameters.get<bool>("output_iteration_info")),
+    _relative_tolerance(parameters.get<Real>("relative_tolerance")),
+    _absolute_tolerance(parameters.get<Real>("absolute_tolerance")),
+    _line_search(true),
+    _bracket_solution(true),
+    _check_range(false)
+{
+  if (parameters.get<bool>("legacy_return_mapping") == true)
+  {
+    if (!parameters.isParamSetByUser("relative_tolerance"))
+      _relative_tolerance = 1.e-5;
```

Comment:
In general, the default was 1e-5, but there is a BISON model that actually set it looser.  I have retained that behavior in my BISON code that accompanies this.